jsFiles/utility.js

```diff
@@ -0,0 +1,73 @@
+// displayWeatherUserLocation() is used to get the location of the user's computer: latitude and longitude
+//(geographical coordinates). These data is used to build the url for Open Weather API,
+// where the data of the weather data is fetch from.
+function displayWeatherUserLocation(pos) {
+  const latitude = pos.coords.latitude.toFixed(2);
+  const longitude = pos.coords.longitude.toFixed(2);
+  const apiUrl = `${apiRootUrl}weather?lat=${latitude}&lon=${longitude}&appid=${KEY}&units=metric`;
+  fetchData(apiUrl);
+}
+
+// function call by submit event from form
+// takes the city input, builds api url and call fetch
+function search(event) {
+  event.preventDefault();
+  //The preventDefault() method cancels the event if it is cancelable, 
+  // meaning that the default action that belongs to the event will not occur.
+  // submiting a form is cancelled
+  //prevent the default action of the form (reloading the page)
+  let cityRequested = inputElement.value;
+  const api = `${apiRootUrl}weather?q=${cityRequested}&appid=${KEY}&units=metric`;
+  fetchData(api);
+}
+
+
+// displays date and time
+// TODO: fix for when displaying the time of a searched city -- timezones offset
+function dateTimeDisplay(dateObj) {
+  let date;
+  let time;
+  date = `${dateObj.getDate()}/${
```

Comment:
but actually you can do it like that:
```js
  const date = `${dateObj.getDate()}/${
    dateObj.getMonth() + 1
  }/${dateObj.getFullYear()}`;
```

Same for time. Because you don't changing those variables so you can make them constant.